server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -653,13 +653,6 @@ public Iterator<Setting<?>> settings() {
         Property.Final
     );
 
-    public static final Setting<Boolean> SYNTHETIC_SOURCE_SECOND_DOC_PARSING_PASS_SETTING = Setting.boolSetting(
-        "index.synthetic_source.enable_second_doc_parsing_pass",
-        true,
-        Property.IndexScope,
-        Property.Dynamic
-    );
```

Comment:
Not sure if it's safe to remove a setting, would this cause issues with rolling upgrades? It shouldn't be set anywhere, of course.

Reply:
Shard of indices using setting would not be able allocate if this setting is used, but it has been removed. That what branches did this setting get back ported? I think just removing the setting, should fix the issue, but would be good to double check this.

Reply:
This was added to 8.16, not used anywhere as it's a last-resort setting: https://github.com/elastic/elasticsearch/pull/114649

Reply:
I see. I'm not against removing this. I was trying to quantify the severity of the issue if someone ends up using this setting and then upgrades.